docs/guides/python-collector.md

```diff
@@ -411,11 +411,20 @@ min/max/average temperature chart with multiple dimensions:
 
```

Comment:
```suggestion
## Add a configuration file

The last piece of the puzzle to create a fully robust python collector is the configuration file. Python.d uses configuration in [YAML](https://www.tutorialspoint.com/yaml/yaml_basics.htm) format and is used as follows:

- Create a configuration file in the same directory as the `<plugin_name>.chart.py`. Name it `<plugin_name>.conf`.
- Define a `job`, which is an instance of the collector. It is useful when you want to collect data from different sources with different attributes. For example, we could gather data from 2 different weather stations, which use different temperature measures: Fahrenheit and Celcius.
- You can define many different jobs with the same name, but with different attributes. Netdata will try each job serially and will stop at the first job that returns data. If multiple jobs have the same name, only one of them can run. This enables you to define different "ways" to fetch data from a particular data source so that the collector has more chances to work out-of-the-box. For example, if the data source supports both `HTTP` and `linux socket`, you can define 2 jobs named `local`, with each using a different method. 
- Check the `postgresql` collector configuration file on [GitHub](https://github.com/netdata/netdata/blob/master/collectors/python.d.plugin/postgres/postgres.conf) to get a sense of the structure.

```yaml
weather_station_1:
    name: 'Greece'
    endpoint: 'https://endpoint_1.com'
    port: 67
    type: 'celcius'
weather_station_1:
    name: 'Florida: USA'
    endpoint: 'https://endpoint_2.com'
    port: 67 
    type: 'fahrenheit'
   ```  
   
Then,  you can access the above configuration variables in the `__init__` function:

```python
def __init__(self, configuration=None, name=None):
        SimpleService.__init__(self, configuration=configuration, name=name)
        self.endpoint = self.configuration.get('endpoint', <default_endpoint>)
```
It's worth noting that since you initiate the `framework class`,e.g `SimpleService.__init__`, the configuration will be available throughout the whole `Service` class of your module, as `self.configuration`. Finally, note that the `configuration.get` function takes 2 arguments, one with the name of the configuration field and one with a default value in case it doesn't find the configuration field. This allows you to define sane defaults for your collector.

Moreover, when creating the configuration file, create a large comment section that describes the configuration variables and inform the user about the defaults. For example, take a look at the `postgresql` collector on [GitHub](https://github.com/netdata/netdata/blob/master/collectors/python.d.plugin/postgres/postgres.conf).

You can read more about the configuration file on the [Python.d documentation](https://learn.netdata.cloud/docs/agent/collectors/python.d.plugin). 
```

Reply:
I added a mention about configuration files @joelhans  and @andrewm4894 . What do you think? I feel that I covered all the gotchas, but please feel free to suggest additions!

Reply:
i like this - a nice example of multiple jobs and params. i think it's could be `weather_station_gr` and `weather_station_us` as the job names. Or at least looks like the both `weather_station_1` in the suggestion, maybe you meant _1 and _2? Either way, i think the names need to be unique for each job.

Reply:
it would be nice to then mention or link to the fully done example collector in a repo somewhere that then uses the `type` param in the chart definition maybe. 

Or maybe we should just make a small PR to add a bit more to `example.conf` and then just link to that as another nice complete canonical example: https://github.com/netdata/netdata/blob/master/collectors/python.d.plugin/example/example.conf

I am actually going to make a PR to add some jobs and params to the example collector as it really should have some. https://github.com/netdata/netdata/tree/master/collectors/python.d.plugin/example

Reply:
I mention the postgresql example for the different jobs-same name example. You think I should add the example.conf as well and underline how the attribute can surface as a chart attribute?

(Nice PR on the example btw ✌️)

Reply:
i made this PR so the example collector can be referenced when discussing jobs and params if useful in this guide: https://github.com/netdata/netdata/pull/10777

Reply:
@OdysLam oh  i mean i'm just not sure this will work:

![image](https://user-images.githubusercontent.com/2178292/111313396-376bfc80-8658-11eb-97d0-8a8cd380d8dc.png)

I don't think they can both be called `weather_station_1`. Unless maybe they can? But not sure. I think not allowed from reading [this](https://github.com/netdata/netdata/blob/master/collectors/python.d.plugin/example/example.conf#L43)

Reply:
Fixed it @andrewm4894 , thanks! You were absolutely correct.

Reply:
Nice work @OdysLam, thanks for adding this!